Python/errors.c

```diff
@@ -292,6 +296,10 @@ PyErr_NormalizeException(PyObject **exc, PyObject **val, PyObject **tb)
 finally:
     Py_DECREF(type);
     Py_DECREF(value);
+    if (recursion_depth + 1 == Py_NORMALIZE_RECURSION_LIMIT) {
+        PyErr_Format(PyExc_RecursionError, "maximum recursion depth exceeded "
```

Comment:
A nit, but you can use `PyErr_SetString` if you don't use the formatting capabilities.